src/scores/probability/fss_impl.py

```diff
@@ -0,0 +1,528 @@
+"""
+This module contains methods related to the FSS score
+
+For an explanation of the FSS, and implementation considerations,
+see: [Fast calculation of the Fractions Skill Score][fss_ref]
+
+[fss_ref]:
+https://www.researchgate.net/publication/269222763_Fast_calculation_of_the_Fractions_Skill_Score
+"""
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+import xarray as xr
+
+from scores import utils
+from scores.typing import XarrayLike
+
+# Note: soft keyword `type` only support from >=3.10
+# "tuple" of 64 bit floats
+f8x3 = np.dtype("f8, f8, f8")
+# Note: `TypeAlias` on variables for python <=3.9
+if TYPE_CHECKING:  # pragma: no cover
```

Comment:
Current Python testing is on 3.9 and up, so I think you can drop support for earlier versions.

Reply:
I meant to say 3.9 and below, not sure why I put <=3.9, <3.10 is what I wanted to say.

Reply:
As an aside, 3.10 has a lot of improvements in the typing libs, any reason not to bump the support to >=3.10?

Reply:
There was a period where Python going from 3.9 to 3.10 broke a bunch of Linux systems and tools which assumed that version numbers only had one decimal place and lexical ordering. It's an older problem, but not that old. I suspect it's probably okay, but that's why. On the other hand, 3.10 was released in October 2021. I haven't had any complaints about version support to date, so I'm just trying to be conservative. I had some other features which also transition at 3.10 which would be nice to be able to focus on. I will raise the question with the other contributors and see what they thing.

Reply:
Makes sense. I'll just change the comment to <3.10, to reflect the compatibility reason.

Reply:
I did a bit of extra checking on use cases. Python 3.9 remains a requirement unless we're absolutely unable to maintain compatibility in future.